In [7]:
#finding the mean,median and mode for the array data
import numpy as np
from scipy import stats

data = np.array([2,3,4,5,6,7,8,9,100])
dt_mean = np.mean(data)
dt_median = np.median(data)
dt_mode = stats.mode(data)
print(dt_mean)
print(dt_median)
print(dt_mode[0][0])

16.0
6.0
2


In [8]:
from statistics import variance,stdev
game_points = np.array([24,2,5,25,34,64,56,78,90,42,67])
#calculating variance
dt_var = variance(game_points)
print(dt_var)

849


In [9]:
#calculating standard deviation
dt_stdev = stdev(game_points)
print(dt_stdev)

29.13760456866693


In [10]:
#calculating range
dt_rnge = np.max(game_points) - np.min(game_points)
print(dt_rnge)

88


In [13]:
print("quantiles")
for i in [20,60,80]:
    dt_qntls = np.percentile(game_points,i)
    print(i,"% :",dt_qntls)

quantiles
20 % : 24.0
60 % : 56.0
80 % : 67.0


In [16]:
Q1 = np.percentile(game_points,25)
Q3 = np.percentile(game_points,75)
IQR = Q3 - Q1
print(Q1,Q3,IQR)

24.5 65.5 41.0


# Example of hypothesis testing:
A chocolate manufacturer who is also your friend claims that all chocolates produced from his factory weigh at least 1,000 g and you have got a funny feeling that it might not be true; you both collected a sample of 30 chocolates and found that the average chocolate weight as 990 g with sample standard deviation as 12.5 g. Given the 0.05 significance level, can we reject the claim made by your friend?

The null hypothesis is that µ0 ≥ 1000 (all chocolates weigh more than 1,000 g).


In [19]:
# we are solving this problem by finding out the test statistic and comparing it with critical t-value
from scipy import stats
xbar = 990; mu0 = 1000; s = 12.5; n=30
#xbar is avg weight of chocolate in n sample i.e 30
#mu0 is assumption made by the person that weigh is greater than 1000
#s is the standard deviation of the data
#n is the size of the data
t_sample = (xbar-mu0)/(s/np.sqrt(n))
print("test statistic : ",t_sample)

test statistic :  -4.381780460041329


In [22]:
alpha = 0.05
#alpha is the significance level of the hyphothesis
#using that alpha value we are trying to find out the t-critical value 
t_alpha = stats.t.ppf(alpha,n-1)
print("critical value from t-table: ",t_alpha)

critical value from t-table:  -1.6991270265334977


In [23]:
#now we are trying to find out the minium p_val to compare with t-critical 
p_val = stats.t.sf(np.abs(t_sample),n-1)
print(p_val)

7.035025729010886e-05


# Final hypothesis: Test statistic is -4.3818, which is less than the critical value of -1.699. Hence, we can reject the null hypothesis (your friend's claim) that the mean weight of a chocolate is above 1,000 g.

# 2nd Example: Assume that the test scores of an entrance exam fit a normal distribution. Furthermore, the mean test score is 52 and the standard deviation is 16.3. What is the percentage of students scoring 67 or more in the exam?

In [25]:
from scipy import stats
xbar = 67; mu0 = 52; s = 16.3
# Calculating z-score
z = (67-52)/16.3
# Calculating probability under the curve
p_val = 1- stats.norm.cdf(z)
print ("Prob. to score more than 67 is",round(p_val*100,2),"%")


Prob. to score more than 67 is 17.87 %


In [2]:
import pandas as pd
from scipy import stats

survey = pd.read_csv("survey.csv")
survey.head()

Sex  Wr.Hnd  NW.Hnd  W.Hnd     Fold  Pulse     Clap  Exer  Smoke  \
0  Female    18.5    18.0  Right   R on L   92.0     Left  Some  Never   
1    Male    19.5    20.5   Left   R on L  104.0     Left  None  Regul   
2    Male    18.0    13.3  Right   L on R   87.0  Neither  None  Occas   
3    Male    18.8    18.9  Right   R on L    NaN  Neither  None  Never   
4    Male    20.0    20.0  Right  Neither   35.0    Right  Some  Never   

   Height       M.I     Age  
0   173.0    Metric  18.250  
1   177.8  Imperial  17.583  
2     NaN       NaN  16.917  
3   160.0    Metric  20.333  
4   165.0    Metric  23.667

In [4]:
survey_tab = pd.crosstab(survey.Smoke,survey.Exer,margins = True)
observed = survey_tab.iloc[0:4,0:3]

In [6]:
#we have created a table for smokers
observed

Exer   Freq  None  Some
Smoke                  
Heavy     7     1     3
Never    87    18    84
Occas    12     3     4
Regul     9     1     7

In [5]:
contg = stats.chi2_contingency(observed= observed)
p_value = round(contg[1],3)
print ("P-value is: ",p_value)

P-value is:  0.483
